google/cloud/bigtable/row_response.py

```diff
@@ -91,20 +329,51 @@ def __init__(
         value: row_value,
         row: row_key,
         family: family_id,
-        column_qualifier: qualifier,
+        column_qualifier: qualifier | str,
+        timestamp_ns: int,
         labels: list[str] | None = None,
-        timestamp: int | None = None,
     ):
+        """
+        CellResponse constructor
+
+        CellResponse objects are not intended to be constructed by users.
+        They are returned by the Bigtable backend.
+        """
         self.value = value
         self.row_key = row
         self.family = family
+        if isinstance(column_qualifier, str):
+            column_qualifier = column_qualifier.encode()
         self.column_qualifier = column_qualifier
-        self.labels = labels
-        self.timestamp = timestamp
+        self.timestamp_ns = timestamp_ns
+        self.labels = labels if labels is not None else []
 
-    def decode_value(self, encoding="UTF-8", errors=None) -> str:
-        """decode bytes to string"""
-        return self.value.decode(encoding, errors)
+    @staticmethod
+    def _from_dict(
+        row_key: bytes, family: str, qualifier: bytes, cell_dict: dict[str, Any]
+    ) -> CellResponse:
+        """
+        Helper function to create CellResponse from a dictionary
+
+        CellResponse objects are not intended to be constructed by users.
+        They are returned by the Bigtable backend.
+        """
+        # Bigtable backend will use microseconds for timestamps,
+        # but the Python library prefers nanoseconds where possible
+        timestamp = cell_dict.get(
+            "timestamp_ns", cell_dict.get("timestamp_micros", -1) * 1000
```

Comment:
what's the benefit of converting it back to nano seconds?

Reply:
Python's time module returns time as nanoseconds or (float) seconds, but the Bigtable backend returns time as microseconds. That makes things a bit complicated, since conversion will be required somewhere. I opted to keep the library's canonical representation in nanos, and convert to micros when needed

Since we have these two time representations, I made the `from_dict` function support both, but it still needs to convert it into the canonical representation when building the object

Reply:
Gotcha, I'm ok with doing the conversion in the implementation, but we should be careful not to expose nano precision on the public surface because a customer could try set the timestamp with nano seconds precision.